File: leetcode/1881.go
package leetcode

func maxValue(n string, x int) string {
	X := byte('0' + x)
	i, lens := 0, len(n)
	if n[i] == '-' {
		for ; i < lens && n[i] <= X; i++ {
		}
	} else {
		for ; i < lens && n[i] >= X; i++ {
		}
	}
	return n[:i] + string(X) + n[i:]
}

File: leetcode/1881.cpp
class Solution {
public:
    string maxValue(string s, int x) {
        auto X = x + '0';
        int i = 0, n = s.size();
        if (s[i] == '-') {
            for (; i < n && s[i] <= X; ++i)
                ;
        } else {
            for (; i < n && s[i] >= X; ++i)
                ;
        }
        s.insert(i, 1, X);
        return s;
    }
};